src/main/kotlin/calculator/Operator.kt

```diff
@@ -0,0 +1,23 @@
+package calculator
+
+enum class Operator(private val op: String, private val operationLambda: (leftOperand: Double, rightOperand: Double) -> Double) {
+
+    ADDITION("+", { leftOperand, rightOperand -> leftOperand + rightOperand }),
+    SUBTRACTION("-", { leftOperand, rightOperand -> leftOperand - rightOperand }),
+    MULTIPLICATION("*", { leftOperand, rightOperand -> leftOperand * rightOperand }),
+    DIVISION("/", { leftOperand, rightOperand ->
+        check(rightOperand != 0.0) { "0 으로 나눌 수 없습니다." }
+        leftOperand / rightOperand
+    });
+
+    companion object {
+
+        private val OPERATORS_BY_OP = values().associateBy { it.op }
+
+        fun get(operator: String) = let {
+            OPERATORS_BY_OP[operator] ?: throw IllegalArgumentException("연산 불가능한 연산자가 입력되었습니다.: $operator")
+        }
```

Comment:
`associateBy`가 매번 호출될때마다 `Map`객체를 생성하게 될 것 같네요!
간단하게 아래와같이 변경할 수도 있습니다!

```suggestion
        fun get(operator: String) = values().find { it.op == operator }
            ?: throw IllegalArgumentException("연산 불가능한 연산자가 입력되었습니다.: $operator")
```